concurrent.lib.sh

```diff
@@ -144,6 +144,12 @@ concurrent() (
     __crt__ORIG_BASHOPTS=${BASHOPTS}
     __crt__ORIG_SHELLOPTS=${SHELLOPTS}
 
+    if [[ `uname` == 'Darwin' ]]; then
+        __crt__sed='gsed'   # for OS X
+    else
+        __crt__sed='sed'
+    fi
+
```

Comment:
This wouldn't work on some OS X systems (like mine!) where `gsed` is installed as `sed`. What do you think about this instead:

```bash
    if sed --version &> /dev/null; then  # BSD sed has no --version
        __crt__sed='sed'
    elif gsed --version &> /dev/null; then
        __crt__sed='gsed'
    else
        __crt__error "Requires GNU sed"
    fi
```

I also recommend moving this check into the "Compatibility Check" section, under the bash version check.